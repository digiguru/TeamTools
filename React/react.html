<html>
    <head>

           <link rel="stylesheet" href="../Shared/styles.css">

    </head> 
        
    
    <body id="stage" class="coloured-bg">
        <div id="tuckman"></div> 
        <div id="container"></div>
        <div id="comfort"></div>

    </body>
    <script src="../3rdParty/require.min.js" data-main="js/compiled"></script>
</html>