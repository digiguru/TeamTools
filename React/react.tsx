/*
<ul id="users"><li><span class="user">Adam Hall</span><a href="void(0);">X</a></li><li><span class="user">Billie Davey</span><a href="void(0);">X</a></li><li><span class="user">Laura Rowe</span><a href="void(0);">X</a></li><li><span class="user">Simon Dawson</span><a href="void(0);">X</a></li><li><span class="user">Fred</span><a href="void(0);">X</a></li></ul>
*/
class Delete extends React.Component<any, any> {
    render() {
        return <a href="void(0);">X</a>;
    }
}
class User extends React.Component<any, any> {
  render() {
    return <li><span className="user">{this.props.username}</span><Delete /></li>;
  }
}
class UserList extends React.Component<any, any> {
    render() {
        let users = [];

        this.props.users.forEach(function(user) {
            users.push(<User user={user} username={user.name} key={user.name} />);
        });
        return <ul id="users">{users}</ul>;
    }
}
let USERS = [
    {name: "bob"},
    {name: "Donald"}
];
ReactDOM.render(
  <UserList users={USERS} />,
  document.getElementById("container")
);