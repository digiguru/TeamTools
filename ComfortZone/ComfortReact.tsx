import * as React from "react";

const 
    styleChaos = {
        fill: "rgb(0, 180, 219)"
    },
    styleStretch = {
        fill: "rgb(0, 215, 254);"
    },
    styleComfot = {
        fill: "rgb(0, 215, 254);"
    }

export default class ComfortReact extends React.Component<any,any> {

  render() {
    return <svg id="stage" width="800" height="800">
            <rect className="area js-area-standard" id="chaos" width="800" height="800" style="{styleChaos}"></rect>


            <g id="zones">
                <circle className="area js-area-standard" id="stretch" r="300.12599083141845" cx="400" cy="400"  style="{styleStretch}"></circle>
                <circle className="area js-area-standard" id="comfort" r="97.516770592797" cx="400" cy="400" style="{styleComfort}"></circle>
            </g>
            <g id="history">
            </g>
            <text className="area-label" id="label-choas" text-anchor="middle" x="400" y="50">chaos</text>
            <text className="area-label" id="label-stretch" text-anchor="middle" x="400" y="200">stretch</text>
            <text className="area-label" id="label-comfort" text-anchor="middle" x="400" y="400">comfort</text>
            <rect id="clickable" width="800" height="800" fill-opacity="0.0"></rect>
        </svg>;
  }

}