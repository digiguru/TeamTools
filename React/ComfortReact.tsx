import * as React from "react";

export interface IResizableModel {
    width: string;
    height: string;
}
export interface IInteractiveModelState {
    focus: string;
}
export interface IResizableInteractiveModelState extends IInteractiveModelState  {
    focus: string;
    width: string;
    height: string;
}

export class ChaosArea extends React.Component<any, IResizableInteractiveModelState> {
     constructor(props) {
        super(props);
        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);
        this.state = {
            focus: "not-in-focus",
            width: props.width || "100%",
            height: props.height || "100%"
        };
    }

    _onMouseEnter() {
        this.setState({focus: "in-focus"});
    }

    _onMouseLeave() {
        this.setState({focus: "not-in-focus"});
    }

    render() {
        const className: string = this.state.focus + " area js-area-standard";
        return <g>
            <rect className={className} onMouseEnter={this._onMouseEnter} onMouseLeave={this._onMouseLeave} id="chaos" width={this.state.width} height={this.state.height}></rect>
            <text className="area-label" id="label-choas" text-anchor="middle" textAnchor="middle" x="50%" y="20">chaos</text>
        </g>;
    }
}

export class StretchArea extends React.Component<any, IResizableInteractiveModelState> {
    constructor(props) {
        super(props);

        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {
            focus: "not-in-focus",
            width: props.width || "100%",
            height: props.height || "100%"
        };
    }

    _onMouseEnter() {
        this.setState({focus: "in-focus"});
    }

    _onMouseLeave() {
        this.setState({focus: "not-in-focus"});
    }

    render() {
        const className: string = this.state.focus + " area js-area-standard";
        return <g>
            <circle 
                onMouseEnter={this._onMouseEnter} 
                onMouseLeave={this._onMouseLeave} 
                className={className} id="stretch" r="0%" cx="50%" cy="50%">
                {this.props.children}
            </circle>
            <text className="area-label" id="label-stretch" text-anchor="middle" textAnchor="middle" x="50%" y="20%">stretch</text>
        </g>;
    }
}
export class BouncyAnimation extends React.Component<any, any> {
     render() {
         const delay = this.props.delay || "0s";
         const duration = this.props.duration || "0.8s";
         const toValue = parseInt(this.props.value || (20), 10);
         const toValueType = this.props.valueType || "%";
         const values = [
            0 + toValueType,
            (toValue + toValue / 4) + toValueType,
            (toValue - toValue / 10) + toValueType,
            (toValue + toValue / 20) + toValueType,
            (toValue) + toValueType,
         ];
         const valuesToString = values.join(";");
         return <animate attributeType="XML" attributeName="r" from="0%" to="20%" 
                    dur={duration} 
                    begin={delay} 
                    values={valuesToString}
                    keyTimes="0; 0.3; 0.6; 0.8; 1"
                    fill="freeze"
                    />;
     }
}
export class ComfortArea extends React.Component<any, IResizableInteractiveModelState> {
     constructor(props) {
        super(props);

        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {
            focus: "not-in-focus",
            width: props.width || "100%",
            height: props.height || "100%"
        };
    }

    _onMouseEnter() {
        this.setState({focus: "in-focus"});
    }

    _onMouseLeave() {
        this.setState({focus: "not-in-focus"});
    }

    render() {
        const className: string = this.state.focus + " area js-area-standard";
        //const startValue = 0 || this.props.value;
        const value = this.props.value || 20;
        return <g>
            <circle onMouseEnter={this._onMouseEnter}
                onMouseLeave={this._onMouseLeave}
                className={className} id="comfort" r="0%" cx="50%" cy="50%">
                {this.props.children}
            </circle>
            <text className="area-label" id="label-comfort" text-anchor="middle" textAnchor="middle" x="50%" y="50%">comfort</text>
        </g>;
        // keySplines=".42 0 1 1;0 0 .59 1;.42 0 1 1;0 0 .59 1;.42 0 1 1;0 0 .59 1;.42 0 1 1;0 0 .59 1;" 
    
    }
}
export class Stage extends React.Component<any, any> {

    constructor(props) {
        super(props);
        this.state = {
            width: "800",
            height: "800"
        };
    }

    render() {
        return <svg id="stage" width={this.state.width} height={this.state.height}>{this.props.children}</svg>;
    }
}
export class ComfortReact extends React.Component<any, any> {

    render() {
        return <Stage>
                    <ChaosArea />
                    <g id="zones">
                        <StretchArea><BouncyAnimation value="45" /></StretchArea>
                        <ComfortArea><BouncyAnimation value="20" delay="0.5s" /></ComfortArea>
                    </g>
                    <g id="history">
                    </g>
                </Stage>;
                //         <rect id="clickable" width="100%" height="100%" fill-opacity="0.0"></rect>
            
    }

}