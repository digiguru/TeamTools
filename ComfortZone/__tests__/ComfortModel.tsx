import * as React from 'react';
import ComfortReact from '../ComfortReact';
const renderizer = require('react-test-renderer');

it('Should show the comfort model', () => {
    const component = renderizer.create(
        <ComfortReact></ComfortReact>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});