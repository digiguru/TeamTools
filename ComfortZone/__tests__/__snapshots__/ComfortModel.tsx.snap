exports[`test Should show the comfort model 1`] = `
<svg
  height="800"
  id="stage"
  width="800">
  <g
    fill-opacity="0.0"
    id="assets">
    <path
      d="m216.77742,285.47641l89.687332,115.132935c45.697845,-103.041046 101.639099,-197.834396 191.554749,-287.832077c-67.294983,42.004333 -141.475403,121.768204 -204.841431,220.466995l-76.40065,-47.767853z"
      fill="#007f00"
      id="asset-tick"
      stroke="#007f00"
      stroke-miterlimit="4"
      stroke-width="0" />
  </g>
  <rect
    className="area js-area-standard"
    height="800"
    id="chaos"
    style="{styleChaos}"
    width="800" />
  <g
    id="zones">
    <circle
      className="area js-area-standard"
      cx="400"
      cy="400"
      id="stretch"
      r="300.12599083141845"
      style="{styleStretch}" />
    <circle
      className="area js-area-standard"
      cx="400"
      cy="400"
      id="comfort"
      r="97.516770592797"
      style="{styleComfort}" />
  </g>
  <g
    id="history" />
  <text
    className="area-label"
    id="label-choas"
    text-anchor="middle"
    x="400"
    y="50">
    chaos
  </text>
  <text
    className="area-label"
    id="label-stretch"
    text-anchor="middle"
    x="400"
    y="200">
    stretch
  </text>
  <text
    className="area-label"
    id="label-comfort"
    text-anchor="middle"
    x="400"
    y="400">
    comfort
  </text>
  <rect
    fill-opacity="0.0"
    height="800"
    id="clickable"
    width="800" />
</svg>
------------HTML PREVIEW---------------
<svg
  height="800"
  id="stage"
  width="800"
>
  <g
    fill-opacity="0.0"
    id="assets"
  >
    <path
      d="m216.77742,285.47641l89.687332,115.132935c45.697845,-103.041046 101.639099,-197.834396 191.554749,-287.832077c-67.294983,42.004333 -141.475403,121.768204 -204.841431,220.466995l-76.40065,-47.767853z"
      fill="#007f00"
      id="asset-tick"
      stroke="#007f00"
      stroke-miterlimit="4"
      stroke-width="0"
    >
    </path>
  </g>
  <rect
    class="area js-area-standard"
    height="800"
    id="chaos"
    style="{styleChaos}"
    width="800"
  >
  </rect>
  <g id="zones">
    <circle
      class="area js-area-standard"
      cx="400"
      cy="400"
      id="stretch"
      r="300.12599083141845"
      style="{styleStretch}"
    >
    </circle>
    <circle
      class="area js-area-standard"
      cx="400"
      cy="400"
      id="comfort"
      r="97.516770592797"
      style="{styleComfort}"
    >
    </circle>
  </g>
  <g id="history"></g>
  <text
    class="area-label"
    id="label-choas"
    text-anchor="middle"
    x="400"
    y="50"
  >
    chaos
  </text>
  <text
    class="area-label"
    id="label-stretch"
    text-anchor="middle"
    x="400"
    y="200"
  >
    stretch
  </text>
  <text
    class="area-label"
    id="label-comfort"
    text-anchor="middle"
    x="400"
    y="400"
  >
    comfort
  </text>
  <rect
    fill-opacity="0.0"
    height="800"
    id="clickable"
    width="800"
  >
  </rect>
</svg>
`;
